src/main/scala/treadle/executable/ExpressionCompiler.scala

```diff
@@ -387,28 +387,51 @@ class ExpressionCompiler(
       val arg1Width = getWidth(expressions.head)
       val param1 = ints.head.toInt
 
-      arg1 match {
-        case e1: IntExpressionResult =>
-          op match {
-            case Head => HeadInts(e1.apply _, takeBits = param1, arg1Width)
-            case Tail => TailInts(e1.apply _, toDrop = param1, arg1Width)
-            case Shl  => ShlInts(e1.apply _, GetIntConstant(param1).apply _)
-            case Shr  => ShrInts(e1.apply _, GetIntConstant(param1).apply _)
-          }
-        case e1: LongExpressionResult =>
-          op match {
-            case Head => HeadLongs(e1.apply _, takeBits = param1, arg1Width)
-            case Tail => TailLongs(e1.apply _, toDrop = param1, arg1Width)
-            case Shl  => ShlLongs(e1.apply _, GetLongConstant(param1).apply _)
-            case Shr  => ShrLongs(e1.apply _, GetLongConstant(param1).apply _)
-          }
-        case e1: BigExpressionResult =>
-          op match {
-            case Head => HeadBigs(e1.apply _, takeBits = param1, arg1Width)
-            case Tail => TailBigs(e1.apply _, toDrop = param1, arg1Width)
-            case Shl  => ShlBigs(e1.apply _, GetBigConstant(param1).apply _)
-            case Shr  => ShrBigs(e1.apply _, GetBigConstant(param1).apply _)
-          }
+      if(op == Shl) {
+        (DataSize(getWidth(tpe)), arg1) match {
+          case (IntSize, e1: IntExpressionResult) =>
+            ShlInts(e1.apply _, GetIntConstant(param1).apply _)
+          case (IntSize, e1: LongExpressionResult) =>
+            ShlLongs(e1.apply, GetLongConstant(param1).apply _)
+          case (IntSize, e1: BigExpressionResult) =>
+            ShlBigs(e1.apply _, GetIntConstant(param1).apply _)
```

Comment:
Why isn't this:
`ShlBigs(e1.apply _, GetBigConstant(param1).apply _)`
similar to the other `case (_, e1: BigExpressionResult) =>`
?

Reply:
The shift amount cannot be greater than 2 ^ 31, so I changed them all to be GetIntConstant